controllers/managedocs_controller.go

```diff
@@ -1392,6 +1403,21 @@ func (r *ManagedOCSReconciler) reconcileCephIngressNetworkPolicy() error {
 	return nil
 }
 
+func (r *ManagedOCSReconciler) reconcileProviderAPIServerNetworkPolicy() error {
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.providerAPIServerNetworkPolicy, func() error {
+		if err := r.own(r.providerAPIServerNetworkPolicy); err != nil {
+			return err
+		}
+		desired := templates.ProviderApiServerNetworkPolicyTemplate.DeepCopy()
+		r.providerAPIServerNetworkPolicy.Spec = desired.Spec
+		return nil
+	})
+	if err != nil {
+		return fmt.Errorf("Failed to update provider api server NetworkPolicy: %v", err)
+	}
+	return nil
+}
```

Comment:
Is there a reason we need a different/new policy for this? can't we just specify this in the ingress/egress policy?

Reply:
What you suggest would work, but I thought this would be better for readability since it matches the existing pattern: there is a network policy for ingressNetworkPolicyName, and a separate one for cephIngressNetworkPolicyName. 

If you would like me to write it the way you suggest, which of these policies should I put the new ingress rule in?

Reply:
Ok, you made your case :) 
let's leave it the way you wrote it